### PR TITLE
Profiles types test improvements

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
This PR improves the tests that are present inside the `x/profiles/types` package following the specifications written inside #481. 

## Checklist
- [ ] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [ ] Updated the documentation. 
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: x/profiles/types/models_app_links_test.go [hunk 1]
```diff
@@ -371,28 +382,29 @@ func TestResult_Validate(t *testing.T) {
 }
 
 func TestResult_Failed__Validate(t *testing.T) {
```
Comment (dadamu) [inline x/profiles/types/models_app_links_test.go:384]:
```suggestion
func TestResult_Failed_Validate(t *testing.T) {
```

File: x/profiles/types/models_app_links_test.go [hunk 2]
```diff
@@ -371,28 +382,29 @@ func TestResult_Validate(t *testing.T) {
...
-	usecases := []struct {
+	testCases := []struct {
 		name      string
 		result    *types.Result
 		shouldErr bool
 	}{
 		{
-			name:      "invalid result returns error",
+			name:      "invalid error returns error",
```
Comment (dadamu) [inline x/profiles/types/models_app_links_test.go:391]: Could it be named with something like `invalid error result returns error`?

File: x/profiles/types/models_app_links_test.go [hunk 3]
```diff
@@ -371,28 +382,29 @@ func TestResult_Validate(t *testing.T) {
...
 			result:    types.NewErrorResult(" "),
 			shouldErr: true,
 		},
 		{
-			name:      "valid result returns no error",
+			name:      "valid error returns no error",
```
Comment (dadamu) [inline x/profiles/types/models_app_links_test.go:396]: Same above.

File: x/profiles/types/account_test.go [hunk 4]
```diff
@@ -270,112 +193,183 @@ func TestProfile_Validate(t *testing.T) {
...
+			if tc.shouldErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
 		})
 	}
 }
+
+func TestProfileSerialization(t *testing.T) {
```
Comment (dadamu) [inline x/profiles/types/account_test.go:312]:
```suggestion
func TestProfile_Serialization(t *testing.T) {
```

File: x/profiles/types/models_app_links_test.go (hunk 2 above)
Comment (RiccardoM) [reply x/profiles/types/models_app_links_test.go:391]: @dadamu This test is to make sure that no invalid error string can be provided, thus returns an error. I don't know how to make it easier to understand :joy:

File: x/profiles/types/models_params_test.go [hunk 5]
```diff
@@ -10,170 +9,163 @@ import (
...
-	nameSurnameParams := types.NewNicknameParams(sdk.NewInt(2), sdk.NewInt(1000))
-	nicknameParams := types.NewDTagParams("^[A-Za-z0-9_]+$", sdk.NewInt(3), sdk.NewInt(30))
-	bioParams := sdk.NewInt(1000)
-
-	params := types.NewParams(nameSurnameParams, nicknameParams, bioParams)
-
-	require.Equal(t, params, types.DefaultParams())
-}
-
 func TestValidateParams(t *testing.T) {
```
Comment (dadamu) [inline x/profiles/types/models_params_test.go:12]:
```suggestion
func TestParams_ValidateParams(t *testing.T) {
```

I think we should follow the same naming format.

File: x/profiles/types/models_app_links_test.go (hunk 2 above)
Comment (dadamu) [reply x/profiles/types/models_app_links_test.go:391]: 😂
How about `invalid error string of result returns error`?

**Review by bragaz — APPROVED** [review]
Everything looks good to me

**Review by dadamu — APPROVED** [review]
Everything looks good to me.

File: x/profiles/types/models_app_links_test.go (hunk 2 above)
Comment (RiccardoM) [inline x/profiles/types/models_app_links_test.go:391]:
```suggestion
			name:      "invalid error string returns error",
```

File: x/profiles/types/models_app_links_test.go (hunk 3 above)
Comment (RiccardoM) [inline x/profiles/types/models_app_links_test.go:396]:
```suggestion
			name:      "valid error string returns no error",
```

File: x/profiles/types/models_app_links_test.go (hunk 2 above)
Comment (RiccardoM) [reply x/profiles/types/models_app_links_test.go]: Changed :+1:

**Pull request merged**